packages/back-end/src/events/handlers/slack/slack-event-handler-utils.ts

```diff
@@ -284,6 +291,75 @@ const buildSlackMessageForExperimentDeletedEvent = (
   };
 };
 
+const buildSlackMessageForExperimentWarningEvent = ({
+  data,
+}: ExperimentWarningNotificationEvent): SlackMessage => {
+  let invalidData: never;
+
+  switch (data.type) {
+    case "auto-update-failed": {
+      const text = `Automatic snapshot creation for ${data.experimentName} failed!`;
+
+      return {
+        text,
+        blocks: [
+          {
+            type: "section",
+            text: {
+              type: "mrkdwn",
+              text:
+                `Automatic snapshot creation for *${data.experimentName}* failed!` +
+                getExperimentUrlFormatted(data.experimentId),
+            },
+          },
+        ],
+      };
+    }
+
+    case "multiple-exposures": {
+      const baseText = `${data.usersCount} users (${data.percent}%) saw multiple variations and were automatically removed from results. Check for bugs in your implementation, event tracking, or data pipeline.`;
```

Comment:
Do we have text formatters in the back-end?

Reply:
(I'll let Jeremy answer your Q)

But here maybe we can remove the remedial action since we don't really have it in the SRM text and should probably keep the places where we have remedial advice limited (e.g. just in the Health Tab and/or results). So I'd consider just deleting the last sentence here.

Reply:
Great idea.